lib/ddtrace/sampling/matcher.rb

```diff
@@ -0,0 +1,53 @@
+module Datadog
+  module Sampling
+    class Matcher
+      # Returns `true` if the span should conforms to this rule, `false` otherwise
+      #
+      # @abstract
+      # @param [Span] span
+      # @return [Boolean]
+      def match?(span)
+        raise NotImplementedError
+      end
+    end
+
+    class SimpleMatcher < Matcher
+      # Returns `true` for case equality (===) with any object
+      MATCH_ALL = Class.new do
+        # DEV: A class that implements `#===` is ~20% faster than
```

Comment:
have a link? this is neat.

Reply:
I tested it locally while developing this class, I can't upload a gist a benchmark snippet.

Reply:
Interesting idea; I could benchmark it and post results.